File: src/tests/fixtures/sample.ts
// Test fixture for parser and reference finder tests

export function testFunction(param: string): string {
  return `Hello ${param}`;
}

export function callerFunction() {
  const result = testFunction('world');
  console.log(result);
  return result;
}

export function anotherCaller() {
  testFunction('test');
  const value = testFunction('value');
  return value;
}

export class TestClass {
  method() {
    testFunction('from class');
  }
}

// Arrow function
export const arrowCaller = () => {
  return testFunction('arrow');
};

// Nested calls
export function nestedCaller() {
  const inner = () => {
    testFunction('nested');
  };
  inner();
}

File: src/tests/fixtures/sample.py
def greet(name):
    message = f"Hello, {name}!"
    print(message)
    return message

class Calculator:
    def add(self, a, b):
        return a + b
    
    def subtract(self, a, b):
        return a - b

def main():
    greet("World")
    calc = Calculator()
    result = calc.add(1, 2)
    print(result)

if __name__ == "__main__":
    main()
